src/lib/methods/base-query.ts

```diff
@@ -122,6 +124,17 @@ export abstract class BaseQuery extends Method {
 		return this;
 	}
 
+	/**
+	 * Configure the retry policy
+	 *
+	 * @param opts - Retry configuration options.
+	 */
+	retry(opts: Options) {
```

Comment:
I guess it would make sense to have support only the number of attempts.

```ts
retry(retries: number);
retry(options: Options);
```

I guess, most of the time the only thing you want to do is change the number of retries.

Maybe even 

```ts
retry(retries: number, options?: Options)
```

Not sure about this one. Feedback is welcome.

Reply:
Will go for number only, seems like the most user-friendly approach here. I thought that providing the full options for the retry strategy might be something that most users would want but in hindsight, it is more something for power users I guess.

Reply:
I think we should support both though... Using overrides makes things easy. In some use cases you might want to adjust the timings etc. So we should definitely expose the full options object.